feat: Prevent need for scrolling in lower screens (Point of Sale)

devdiogenes opened the pull request:
I screens lower than 814px, the browser creates a scroll in the window, that is inconvenient if the user are using a touch screen, for example, as the user already needs to scroll the items section. This pull-requests improves it, preventing the user of scrolling window till a lowest limit.

### Cart - Before:
![image](https://github.com/user-attachments/assets/ec6fed91-68bc-4ce3-84e4-3caf6754dfea)

### Cart - After:
![image](https://github.com/user-attachments/assets/5a086574-890e-40a5-984c-b51875cfbdbe)

### Payment - Before
![image](https://github.com/user-attachments/assets/2a1ad504-4dc4-4d7d-a9ee-6d922e70417d)

### Payment - After
![image](https://github.com/user-attachments/assets/7567d2a7-e925-4d1a-a9d3-97084f7195da)


no-docs

devdiogenes commented:
Waiting review

diptanilsaha reviewed (changes requested):
![image](https://github.com/user-attachments/assets/c68f6ed8-9901-4788-8f9e-21688bbb0311)

Complete Order Button is going out of the `payment-container` on POS Payments.

devdiogenes commented:
> ![image](https://private-user-images.githubusercontent.com/50792171/419393023-c68f6ed8-9901-4788-8f9e-21688bbb0311.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDEzODI3MjIsIm5iZiI6MTc0MTM4MjQyMiwicGF0aCI6Ii81MDc5MjE3MS80MTkzOTMwMjMtYzY4ZjZlZDgtOTkwMS00Nzg4LThmOWUtMjE2ODhiYmIwMzExLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzA3VDIxMjAyMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNmNDU5YmVmYzMyMGE5YWFmOGM5ZmJiZjM0MzBkZTMxNGZkZGYxMzhkZGNhMTVjOGQ5MjUwMDkyNTA3ZDIwMjQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.OUcsK_HaXIhBcppgE8Ks4LVogquwT9iry7z2SZUZaqM)
> 
> Complete Order Button is going out of the `payment-container` on POS Payments.

In this case, it's because payment-container content is too huge. I added a min-height back to 45rem only in this container to fix it, at least until we have a way to reduce the size of the container content.

diptanilsaha reviewed (changes requested):
![image](https://github.com/user-attachments/assets/3691301e-9bb8-49f7-81fb-ff490ea59f6c)

This isn't looking very good. The size of the containers should be equal.

devdiogenes committed:
feat: Prevent need for scrolling in lower screens (Point of Sale)

devdiogenes commented:
> ![image](https://private-user-images.githubusercontent.com/50792171/420565816-3691301e-9bb8-49f7-81fb-ff490ea59f6c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDE2MjQ2NTEsIm5iZiI6MTc0MTYyNDM1MSwicGF0aCI6Ii81MDc5MjE3MS80MjA1NjU4MTYtMzY5MTMwMWUtOWJiOC00OWY3LTgxZmItZmY0OTBlYTU5ZjZjLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMTAlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzEwVDE2MzIzMVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWZkODljYjFmMjU2ODZlMzNiNDY5ZjBhNDI5MjIyZmM2N2ZmY2NlZjZiZWQ2NzM5ZmQyMGRiZmUzM2IxYjJhM2YmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.z9dIBO3WEi-MaYTF65vZj0UdqFwqIdbgzclRTl3_wJo)
> 
> This isn't looking very good. The size of the containers should be equal.

@diptanilsaha I reorganized the payments design. The images is attached to the pull request description. Feel free to let me know if there are more some new thing to fix in my pull-request

diptanilsaha commented:
@devdiogenes Can you share the screenshot of the change you made, especially of the pos payment page? Also, rebase your branch before you push the change.

devdiogenes commented:
@diptanilsaha The branch is already rebased. I organized the payment modes vertically instead of horizontally, and placed in the left side of number keyboard.

I changed the html this way, and changed some details in the CSS
![image](https://github.com/user-attachments/assets/95372f2a-8b71-4062-ae31-ec3a36e88d59)

There are screenshots of the payment page (before and after) in the description of the pull-request

diptanilsaha commented:
@devdiogenes I tried testing your PR locally and I rebased with the latest changes on `develop` branch but I got the following:

![image](https://github.com/user-attachments/assets/f7d4ba3c-8ab0-4bfd-a5c2-1fcd4b8c4e49)

Also, putting the numpad and the mode of payment options side by side reduces the user experience as the POS Fields appear on the left side of the Numpad, so reducing the space for the POS Fields won't look good.

devdiogenes commented:
@diptanilsaha You need to clear-cache before testing.

In my last commit, the POS Fields was organizated above the numpad. Do you have some suggestion about the organization?

diptanilsaha commented:
@devdiogenes thanks, now it's working. Somehow, the CSS file was not building in my local setup.

diptanilsaha commented:
@devdiogenes, I appreciate your efforts, but I think the previous organization was much better. Instead, you can try reducing the size of the Numpad Key and making the `Additional Information` section scrollable. 

Currently, the numpad keys have padding of `--padding-md` but you can try out with `--padding-sm` and test.

I would love to read your thoughts on this.

HarryPaulo commented:
@diptanilsaha Currently, horizontal payment methods are very bad when you have more than 3 payment methods.

Leaving this vertical would be a way to present more payment methods.


https://github.com/user-attachments/assets/de21d29f-a4e4-4818-a56b-f322c574202a


Look at the comparison between the old POS that was removed from ERPNext and the new POS...

In a job that needs to be as fast as possible, the availability of payment methods is very poor.

The first step would be to place the payment methods vertically, the second step would be to make each payment method take up as little space as possible, to make it available in a more practical and quick way.

diptanilsaha commented:
@HarryPaulo I agree with you and you are not alone who have complained about the UX of Mode of Payment selection and I appreciate what @devdiogenes came up with.

My main concern is placing the 'Additional Information' section, which doesn't make it worse than what it is. If you had the opportunity to move it, where would you place it?

HarryPaulo commented:
@diptanilsaha 
![image](https://github.com/user-attachments/assets/0b079014-5f77-4d4d-be4d-4deffdae38f3)

I imagine this would be the best screen allocation.

1. Payment methods.
2. Additional Info
3. Touchscreen Keyboard


The space reserved for additional information will be basically the same, it just changed location, making payment methods more functional.

Another way is: 

![image](https://github.com/user-attachments/assets/486aa6d1-e0a8-4a45-8193-8c5e10cc033c)

1. Additional Info
2. Touchscreen Keyboard
3. Payment methods.

devdiogenes commented:
@diptanilsaha 

I have never used the Additional Information before. I saw in the code that the fields in configured in POS Settings doctype [(reference)](https://github.com/devdiogenes/erpnext/blob/bcb7ca7ebe5a23156d33b7d99e6ea33bcb87637e/erpnext/selling/page/point_of_sale/pos_payment.js#L51), but when I tried to fill the fields, the form fields are locked

![image](https://github.com/user-attachments/assets/bb466338-f024-4422-9801-0a94161b32fe)

I think the additional information section is not even working now

diptanilsaha commented:
@HarryPaulo your solution looks good but this will work for large screens. For small screens, if we try to fit everything and not make it scrollable, it will be cluttered.

diptanilsaha commented:
@HarryPaulo your solution looks good but this will work for large screens. For small screens, if we try to fit everything and not make it scrollable, it will be cluttered.

diptanilsaha commented:
@devdiogenes try selecting fieldname from the dropdown, it will automatically fetch other fields. You can also mark the field as 'Mandatory', so that it will make sure user have to fill the field before completing the order.

diptanilsaha commented:
@devdiogenes I spoke with other members of Frappe and they appreciated your work. However, they have also suggested some changes. So, I am taking this up and will create a PR for the same and will mention you on that PR.

Thanks for your contribution. Keep contributing.

devdiogenes commented:
@diptanilsaha Thank you, I will be waiting for, if God allow me

HarryPaulo commented:
@diptanilsaha thanks!!
